framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1143,6 +1284,40 @@ private boolean handledByValueChecker(AnnotatedTypeMirror type) {
         }
     }
 
+    /** Determines whether a method is the {@code String.length()} method. */
+    boolean isStringLengthMethod(ExecutableElement method) {
+        return method == lengthMethod;
+    }
+
+    /**
+     * Returns a constant value annotation for a length of an array or string type with a constant
+     * value annotation
+     */
+    AnnotationMirror createArrayLengthResultAnnotation(AnnotatedTypeMirror receiverType) {
+        AnnotationMirror arrayAnno = receiverType.getAnnotation(ArrayLen.class);
+        if (arrayAnno != null) {
```

Comment:
I don't think this code was originally yours, but since it's under review I'm going to suggest that this pattern of calling `getAnnotation` and then checking for `null` is an anti-pattern, since it's rather obfuscated. Do you mind changing this method to use `AnnotatedTypeMirror#getAnnotationInHierarchy` to get whatever annotation it is, and then guard each `if` block with a call to `AnnotationUtils.areSameByClass` with the appropriate annotation? I think that would be much cleaner code.